src/components/portfolio/DetailsMainBacktesting.tsx

```diff
@@ -0,0 +1,346 @@
+    // disable button until new result is being displayed
+    dispatch({ type: 'setUpdateDisabled', payload: true });
+    // remove error from last click
+    dispatch({ type: 'setValid', payload: true });
+    if (
+      state.selectedTo < state.selectedFrom ||
+      new Date() < state.selectedTo
+    ) {
+      dispatch({ type: 'setValid', payload: false });
+      dispatch({ type: 'setUpdateDisabled', payload: false });
+    } else {
+      dispatch({ type: 'setBacktesting', payload: undefined });
+      mockFetch(state.selectedFrom, state.selectedTo);
+      dispatch({ type: 'setUpdateDisabled', payload: false });
+    }
+  };
+
+  return (
+    <div className={classes.backtestingWrapper}>
+      <h2 className={classes.subtitle}>
+        {t('portfolio.details.backtesting.subheading')}
+      </h2>
+      <div className={classes.datePicker}>
+        <form className={classes.form} noValidate>
+          <span className={classes.subtitle}>
+            {t('portfolio.details.from')}:
+          </span>
+          <MuiPickersUtilsProvider utils={MomentUtils}>
+            <KeyboardDatePicker
+              InputProps={{ className: classes.datePickerInput }}
+              value={state.selectedFrom}
+              onChange={(date) =>
+                dispatch({
+                  type: 'setFrom',
+                  // if the onchange date is null don't change the state
+                  payload: date ? date.toDate() : state.selectedFrom,
+                })
+              }
+            />
+          </MuiPickersUtilsProvider>
+          <span className={classes.subtitle}>{t('portfolio.details.to')}:</span>
+          <MuiPickersUtilsProvider utils={MomentUtils}>
+            <KeyboardDatePicker
+              InputProps={{
+                className: classes.datePickerInput,
+              }}
+              helperText={
+                state.inputValid
+                  ? undefined
+                  : t('portfolio.details.backtesting.dateHelperText')
+              }
+              error={!state.inputValid}
+              value={state.selectedTo}
+              onChange={(date) =>
+                dispatch({
+                  type: 'setTo',
+                  // if the onchange date is null don't change the state
+                  payload: date ? date.toDate() : state.selectedFrom,
+                })
+              }
+            />
+          </MuiPickersUtilsProvider>
+          <Button
+            className={classes.updateButton}
+            variant="contained"
+            disabled={state.updateDisabled}
+            onClick={() => onClickUpdate()}
+          >
+            {t('portfolio.details.backtesting.updateButton')}
+          </Button>
+        </form>
+      </div>
+      {!state.backtesting && !state.error && (
+        <CircularProgress color="secondary" />
+      )}
+      {/* TODO maybe replace with ternary operator instead of conditional error and conditional backtesting */}
+      {state.error && (
```

Comment:
I think this TODO is obsolete.

```suggestion
      )}
      {state.error && (
```